node/src/main/scala/org/bitcoins/node/SpvNode.scala

```diff
@@ -41,6 +42,52 @@ case class SpvNode(
     this
   }
 
+  var dataMessageHandler: DataMessageHandler = _dataMessageHandler
+
+  override def updateDataMessageHandler(
+      dataMessageHandler: DataMessageHandler): SpvNode = {
+    this.dataMessageHandler = dataMessageHandler
+    this
+  }
+
+  var clients: Vector[P2PClient] = {
```

Comment:
Let's not use `var`, it's a really bad idea as it is not thread safe. Let's keep immutable data structures and update the internals like `updateDataMessageHandler()`